pkg/kubelet/cm/cpumanager/state/state_checkpoint.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"path"
+	"sync"
+)
+
+// cpuManagerCheckpointName is the name of checkpoint file
+const cpuManagerCheckpointName = "cpu_manager_state"
+
+var _ State = &stateCheckpoint{}
+
+type stateCheckpoint struct {
+	sync.RWMutex
```

Comment:
If you create a property rather then embed `sync.RWMutex` then it won't be possible to change mutex state from outside (sort of hermetization, I would say).